File: Domain/News.cs
namespace Domain
{
  public class News
  {
    public int Id { get; set; }
    public string Title { get; set; }
    public string Slug { get; set; }
    public string Description { get; set; }
    public string FullText { get; set; }
    public string ImageUrl { get; set; }
    public bool IsActive { get; set; } = true;
    public bool IsDeleted { get; set; } = false;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public Gallery Gallery { get; set; }
  }
}

File: client-app/src/app/models/news.ts
import { Gallery } from "./gallery";

export interface INews {
  id: number | undefined;
  title: string;
  slug: string;
  description: string;
  fullText: string;
  imageUrl: string;
  isActive: boolean;
  createdAt?: Date | null;
  updatedAt?: Date | null;
  gallery?: Gallery;
}

export class News implements INews {
  constructor(init: NewsFormValues) {
    this.id = init.id!;
    this.title = init.title;
    this.description = init.description;
    this.fullText = init.fullText;
    this.imageUrl = init.imageUrl;
    this.isActive = init.isActive;
    this.slug = init.slug;
  }
  id: number | undefined;
  title: string;
  description: string;
  fullText: string;
  imageUrl: string;
  isActive: boolean;
  slug: string;
  createdAt?: Date | null = null;
  updatedAt?: Date | null = null;
  gallery?: Gallery = { id: 0, galleryImages: [] };
}

export class NewsFormValues {
  id: number | undefined = undefined;
  title: string = "";
  slug: string = "";
  description: string = "";
  fullText: string = "";
  imageUrl: string = "";
  isActive: boolean = true;

  constructor(news?: NewsFormValues) {
    if (news) {
      this.id = news.id;
      this.title = news.title;
      this.description = news.description;
      this.fullText = news.fullText;
      this.imageUrl = news.imageUrl;
      this.isActive = news.isActive;
      this.slug = news.slug;
    }
  }
}
